### PR TITLE
[WIP] ncm-pcs: pacemaker support via pcs

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

**Comment by jrha** [comment]
Why add an empty Changelog? Does something require it?

File: ncm-pcs/src/main/pan/components/pcs/schema.pan [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+type pcs_contraint_order_resource = {
+    'name' : string
+    'action' ? string with match(SELF, '^(start|stop|promote|demote)$')
+};
+
+type pcs_constraint_order = {
+    'source' : pcs_contraint_order_resource
+    'target' : pcs_contraint_order_resource
+    'options' ? string{}
```
Comment (jrha) [inline ncm-pcs/src/main/pan/components/pcs/schema.pan]: No validation on the options?

File: ncm-pcs/src/main/pan/components/pcs/schema.pan [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+};
+
+type pcs_constraint_location = {
+    @{key is resource, value dict of node(s) with their score(s)}
+    'avoids' ? pcs_score{}{}
```
Comment (jrha) [inline ncm-pcs/src/main/pan/components/pcs/schema.pan]: Is there no further validation that can be done to this `dict` of `dict`?

File: ncm-pcs/src/main/pan/components/pcs/schema.pan [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+}
+type pcs_resource = {
+    include pcs_common_device
+    @{resource device standard ([standard[:provider]:]type) (e.g. ocf)}
+    'standard' ? string
+    @{resource device provider ([standard[:provider]:]type) (e.g. heartbeat)}
+    'provider' ? string
+    @{resource device type ([standard[:provider]:]type) (e.g. IPaddr2)
+      or special cases like master}
+    'type' : string
```
Comment (jrha) [inline ncm-pcs/src/main/pan/components/pcs/schema.pan]: No validation on any of these strings?
I assume that at a minimum they should not contain white-space.

**Comment by jrha** [comment]
@stdweird any update?

**Comment by stdweird** [comment]
i am closing this PR, i (or someone else) can always pick it back up when needed

**Pull request closed without merging**